components/renku_data_services/connected_services/core.py

```diff
@@ -20,16 +20,22 @@ def validate_oauth2_client_patch(patch: apispec.ProviderPatch) -> models.OAuth2C
         url=patch.url,
         use_pkce=patch.use_pkce,
         image_registry_url=patch.image_registry_url,
+        oidc_issuer_url=patch.oidc_issuer_url,
     )
 
 
 def validate_unsaved_oauth2_client(clnt: apispec.ProviderPost) -> models.UnsavedOAuth2Client:
     """Validate the the creation of a new OAuth2 Client."""
     if clnt.image_registry_url is not None:
         validate_image_registry_url(clnt.image_registry_url)
+    if clnt.oidc_issuer_url and models.ProviderKind(clnt.kind.value) != models.ProviderKind.generic_oidc:
+        raise errors.ValidationError(
+            message=f"The field 'oidc_issuer_url' can only be set when kind is set to {models.ProviderKind.generic_oidc.value}.",  # noqa E501
+            quiet=True,
+        )
```

Comment:
issue: I think you need another check. If provider kind is oidc and the oidc issuer url is None.

Reply:
Also please run oidc issuer url through url parse and check that we have at least a scheme (i.e. https) and domain name.